common/functions/src/scalars/semi_structureds/json_extract_path_text.rs

```diff
@@ -0,0 +1,121 @@
+// limitations under the License.
+
+use std::fmt;
+
+use common_datavalues::prelude::*;
+use common_exception::ErrorCode;
+use common_exception::Result;
+
+use crate::scalars::semi_structureds::get::extract_value_by_path;
+use crate::scalars::semi_structureds::get::parse_path_keys;
+use crate::scalars::Function;
+use crate::scalars::FunctionContext;
+use crate::scalars::FunctionDescription;
+use crate::scalars::FunctionFeatures;
+
+#[derive(Clone)]
+pub struct JsonExtractPathTextFunction {
+    display_name: String,
+}
+
+impl JsonExtractPathTextFunction {
+    pub fn try_create(display_name: &str, args: &[&DataTypePtr]) -> Result<Box<dyn Function>> {
+        let data_type = args[0];
+        let path_type = args[1];
+
+        if !path_type.data_type_id().is_string() {
+            return Err(ErrorCode::IllegalDataType(format!(
+                "Invalid argument types for function '{}': ({:?}, {:?})",
+                display_name.to_uppercase(),
+                data_type,
+                path_type
+            )));
+        }
+
+        Ok(Box::new(JsonExtractPathTextFunction {
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic().num_arguments(2))
+    }
+}
+
+impl Function for JsonExtractPathTextFunction {
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn return_type(&self) -> DataTypePtr {
+        NullableType::arc(StringType::arc())
+    }
+
+    fn eval(
+        &self,
+        _func_ctx: FunctionContext,
+        columns: &ColumnsWithField,
+        input_rows: usize,
+    ) -> Result<ColumnRef> {
+        let path_keys = parse_path_keys(columns[1].column())?;
+
+        let data_type = columns[0].field().data_type();
+        let mut builder = ColumnBuilder::<VariantValue>::with_capacity(input_rows);
+        if data_type.data_type_id().is_string() || data_type.data_type_id().is_variant() {
+            let serializer = data_type.create_serializer();
+            match serializer.serialize_json_object(columns[0].column(), None) {
+                Ok(values) => {
+                    for v in values {
+                        builder.append(&VariantValue::from(v));
+                    }
+                }
+                Err(e) => return Err(e),
+            }
+        } else {
+            return NullType::arc().create_constant_column(&DataValue::Null, input_rows);
+        }
+        let variant_column = builder.build(input_rows);
+        let column = extract_value_by_path(&variant_column, path_keys, input_rows, false)?;
+
+        // convert VariantColumn to StringColumn
+        let (is_all_null, valids) = column.validity();
+        if is_all_null {
+            return NullType::arc().create_constant_column(&DataValue::Null, input_rows);
```

Comment:
This is inconsistent with `return_type`.

Reply:
I have removed this.